src/validation.cpp

```diff
@@ -2266,7 +2336,7 @@ static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& s
         }
 
         // restore inputs
-        if (!tx.IsCoinBase() && !tx.IsZerocoinSpend() && !tx.IsSigmaSpend() && !tx.IsZerocoinRemint() && !tx.IsLelantusJoinSplit()) { // not coinbases
```

Comment:
I'm afraid this will cause crash or undefined behaviour when disconnecting old blocks. Maybe it's worth to create a function named `HasNoRegularInputs()` that combines all these?

Reply:
done